WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/PostingActivityBlockAnnouncer.kt

```diff
@@ -0,0 +1,115 @@
+package org.wordpress.android.ui.stats.refresh.utils
+
+import android.view.View
+import android.view.accessibility.AccessibilityEvent
+import android.view.accessibility.AccessibilityEvent.TYPE_VIEW_ACCESSIBILITY_FOCUSED
+import android.view.accessibility.AccessibilityNodeInfo.ACTION_CLICK
+import androidx.core.view.AccessibilityDelegateCompat
+import androidx.core.view.accessibility.AccessibilityNodeInfoCompat
+import org.wordpress.android.R
+import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.ActivityItem
+import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.ActivityItem.Box.INVISIBLE
+import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.ActivityItem.Box.VERY_LOW
+import org.wordpress.android.util.AccessibilityUtils
+import org.wordpress.android.viewmodel.ResourceProvider
+import javax.inject.Inject
+
+class PostingActivityBlockAnnouncer
+@Inject constructor(private val resourceProvider: ResourceProvider) {
+    private lateinit var activityItem: ActivityItem
+    private var currentBoxIndex: Int = 0
+    private var currentBlockIndex: Int = 0
+    private lateinit var blockViews: List<View>
+    private var isStatsInPeriod = false
+
+    fun initialize(activityItem: ActivityItem, vararg blockViews: View) {
+        this.activityItem = activityItem
+        this.blockViews = blockViews.toList()
+
+        setupViewDelegate()
+    }
+
+    private fun setupViewDelegate() {
```

Comment:
Most of the logic that does the announcement lives within this function. As reviewers how do you feel about this? I have been thinking and I was wondering if it's worth breaking this up a bit more with some tests as well or since the functionality works and it achieves the goal of announcing each daily stats then it okay. Let me know :)